Description: change data path
 on system install, data path should be in /usr/share, so we have to change it in the script
Author: Nicolas HOUDELOT (nicolas@demosdebs.org)
Last-Update: 2020-05-09
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/common-install-function
+++ b/common-install-function
@@ -80,25 +80,25 @@
 #added in v1.1
 function NSH_InstallRWFileInTmp()
 {
-    cp "./$1.init" "/tmp/${domain}/$2"
+    cp "${NSH_SHAREDIR}/${domain}/$1.init" "/tmp/${domain}/$2"
 }
 
 #added in v1.62
 function NSH_InstallRWDirInTmp()
 {
-    cp -R "./$1.init" "/tmp/${domain}/$2"
+    cp -R "${NSH_SHAREDIR}/${domain}/$1.init" "/tmp/${domain}/$2"
 }
 
 #added in v1.1
 function NSH_CopyFileInTmp()
 {
-    cp "./$1" "/tmp/${domain}/$1"
+     cp "${NSH_SHAREDIR}/${domain}/$1" "/tmp/${domain}/$1"
 }
 
 #added in v1.1
 function NSH_CopyDirInTmp()
 {
-    cp -R "./$1" "/tmp/${domain}"
+    cp -R "${NSH_SHAREDIR}/${domain}/$1" "/tmp/${domain}"
 }
 
 #addid in V1.1 $XDG_CONFIG_HOME
@@ -117,7 +117,7 @@
 	then
 		ln -s $HOME/.config/${domain}/$2  /tmp/${domain}/$2
 	else
-		cp -r ./$1.init $HOME/.config/${domain}/$2
+		cp -r ${NSH_SHAREDIR}/${domain}/$1.init $HOME/.config/${domain}/$2
 		ln -s $HOME/.config/${domain}/$2  /tmp/${domain}/$2
 	fi
 }
@@ -305,7 +305,7 @@
 		rm -rf /tmp/${domain}
 		mkdir /tmp/${domain}
 	fi
-	unzip DH1.zip -d /tmp/${domain}
+	unzip ${NSH_SHAREDIR}/${domain}/DH1.zip -d /tmp/${domain}
 }
 
 #added in v1.8
@@ -317,7 +317,7 @@
 		rm -rf /tmp/${domain}
 		mkdir /tmp/${domain}
 	fi
-	unzip $1 -d /tmp/${domain}
+	unzip ${NSH_SHAREDIR}/${domain}/$1 -d /tmp/${domain}
 }
 
 #added in v1.6
@@ -414,12 +414,12 @@
 
 function NSH_SetDebugDOSBOX()
 {
-	head -n -2 ./${domain}.conf > /tmp/${domain}_debug.conf
+	head -n -2 ${NSH_SHAREDIR}/${domain}/${domain}.conf > /tmp/${domain}_debug.conf
 }
 
 function NSH_SetEXEDOSBOX()
 {
-	head -n -2 ./${domain}.conf > /tmp/${domain}_temp.conf
+	head -n -2 ${NSH_SHAREDIR}/${domain}/${domain}.conf > /tmp/${domain}_temp.conf
 	echo $1 >> /tmp/${domain}_temp.conf
 	echo exit >> /tmp/${domain}_temp.conf
 }
@@ -545,7 +545,7 @@
 function NSH_prepare_std_first_run_WINE()
 {
 	NSH_RemoveShortcutWINE "$APP_FULLNAME"
-	/usr/share/playonlinux/playonlinux-bash ./${domain}_userinstall.pol
+	/usr/share/playonlinux/playonlinux-bash ${NSH_SHAREDIR}/${domain}/${domain}_userinstall.pol
 	NSH_WINE_backupcurrentwine
 	echo ${domain} set to wine $polarch $polwine in $polscreenmode mode
 }
@@ -558,7 +558,7 @@
 		export polscreenmode
 		export polwine
 		NSH_RemoveShortcutWINE "$APP_FULLNAME"
-		/usr/share/playonlinux/playonlinux-bash ./${domain}_userinstall.pol
+		/usr/share/playonlinux/playonlinux-bash ${NSH_SHAREDIR}/${domain}/${domain}_userinstall.pol
 		NSH_WINE_backupcurrentwine
 		echo ${domain} set to wine $polarch $polwine in $polscreenmode mode
 	else
@@ -569,7 +569,7 @@
 			export polscreenmode
 			export polwine
 			NSH_RemoveShortcutWINE "$APP_FULLNAME"
-			/usr/share/playonlinux/playonlinux-bash ./${domain}_userinstall.pol
+			/usr/share/playonlinux/playonlinux-bash ${NSH_SHAREDIR}/${domain}/${domain}_userinstall.pol
 			NSH_WINE_backupcurrentwine
 			echo ${domain} set to wine $polarch $polwine in $polscreenmode mode
 		fi
@@ -630,10 +630,10 @@
 	fi
 	if [ $(mount | grep -c /tmp/cd_${domain}) -eq 0 ]
 	then
-		fuseiso -p ./cdrom/${domain}.iso /tmp/cd_${domain}
+		fuseiso -p ${NSH_SHAREDIR}/${domain}/cdrom/${domain}.iso /tmp/cd_${domain}
 	else
 		fusermount -u /tmp/cd_${domain}
-		fuseiso -p ./cdrom/${domain}.iso /tmp/cd_${domain}
+		fuseiso -p  ${NSH_SHAREDIR}/${domain}/cdrom/${domain}.iso /tmp/cd_${domain}
 	fi
 }
 
@@ -657,10 +657,10 @@
 	fi
 	if [ $(cdemu status | grep -c $1) -eq 0 ]
 	then
-		cdemu load $2 ./cdrom/$1
+		cdemu load $2 ${NSH_SHAREDIR}/${domain}/cdrom/$1
 	else
 		cdemu unload $2
-		cdemu load $2 ./cdrom/$1
+		cdemu load $2 ${NSH_SHAREDIR}/${domain}/cdrom/$1
 	fi
 }
 
@@ -813,10 +813,10 @@
 		echo "[NSH_POL_Create_Install_Link] : removing $WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
 		rm -rf "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
 		echo "[NSH_POL_Create_Install_Link] : creating $INSTALL_DIR in $WINEPREFIX/drive_c/$PROGRAMFILES/"
-		ln -s "$(pwd)" "$WINEPREFIX/drive_c/$PROGRAMFILES/${INSTALL_DIR}"
+		ln -s "${NSH_SHAREDIR}/$INSTALL_DIR" "$WINEPREFIX/drive_c/$PROGRAMFILES/${INSTALL_DIR}"
 	else
 		echo "[NSH_POL_Create_Install_Link] : creating $INSTALL_DIR in $WINEPREFIX/drive_c/$PROGRAMFILES/"
-		ln -s "$(pwd)" "$WINEPREFIX/drive_c/$PROGRAMFILES/${INSTALL_DIR}"
+		ln -s "${NSH_SHAREDIR}/$INSTALL_DIR" "$WINEPREFIX/drive_c/$PROGRAMFILES/${INSTALL_DIR}"
 	fi
 }
 
@@ -828,23 +828,23 @@
 		echo "[NSH_POL_Create_Install_Link] : removing $WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
 		rm -rf "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
 		echo "[NSH_POL_Create_Install_Link] : creating $INSTALL_DIR in $WINEPREFIX/drive_c/$PROGRAMFILES/"
-		ln -s "$(pwd)/hdd" "$WINEPREFIX/drive_c/$PROGRAMFILES/${INSTALL_DIR}"
+		ln -s "${NSH_SHAREDIR}/$INSTALL_DIR/hdd" "$WINEPREFIX/drive_c/$PROGRAMFILES/${INSTALL_DIR}"
 	else
 		echo "[NSH_POL_Create_Install_Link] : creating $INSTALL_DIR in $WINEPREFIX/drive_c/$PROGRAMFILES/"
-		ln -s "$(pwd)/hdd" "$WINEPREFIX/drive_c/$PROGRAMFILES/${INSTALL_DIR}"
+		ln -s "${NSH_SHAREDIR}/$INSTALL_DIR/hdd" "$WINEPREFIX/drive_c/$PROGRAMFILES/${INSTALL_DIR}"
 	fi
 }
 
 
 function NSH_POL_Copy_To_Install_Dir()
 {
-	cp -R . "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
+	cp -R "${NSH_SHAREDIR}/$INSTALL_DIR"/. "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
 }
 
 #added in v1.10
 function NSH_POL_Copy_hdd_To_Install_Dir()
 {
-	cp -R "hdd/." "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
+	cp -R "${NSH_SHAREDIR}/$INSTALL_DIR/hdd/." "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
 }
 
 #updated in v1.39
@@ -875,20 +875,19 @@
 	rm "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR/fake.exe"
 	sed -i -e "s/POL_Wine \"fake.exe\"/POL_Wine start \"$EXE\"/g" "$HOME/.PlayOnLinux/shortcuts/$TITLE"
 	POL_ExtractIcon "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR" "$POL_USER_ROOT/icones/32/"
-	cd "$CURRENT_DIR"
 }
 
 function NSH_POL_Replace_Install_Dir_By_Link()
 {
 	rm -rf "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
-	ln -s "$CURRENT_DIR" "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
+	ln -s "${NSH_SHAREDIR}/$INSTALL_DIR" "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
 }
 
 #added in v1.10
 function NSH_POL_Replace_hdd_Install_Dir_By_Link()
 {
 	rm -rf "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
-	ln -s "$CURRENT_DIR/hdd" "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
+	ln -s "${NSH_SHAREDIR}/$INSTALL_DIR/hdd" "$WINEPREFIX/drive_c/$PROGRAMFILES/$INSTALL_DIR"
 }
 
 #MAME
